source/fundamentals/builders/aggregates.txt

```diff
@@ -234,6 +234,230 @@ of the values of the ``quantity`` field into the ``totalQuantity`` and
+- :ref:`lastN() <java_aggregates_last_n>`
+- :ref:`top() <java_aggregates_top>`
+- :ref:`topN() <java_aggregates_top_n>`
+- :ref:`bottom() <java_aggregates_bottom>`
+- :ref:`bottomN() <java_aggregates_bottom_n>`
+
+.. tip::
+
+   You can only perform aggregation operations with these pick-n accumulators
+   when running MongoDB v5.2 or later.
+
+Learn which aggregation pipeline stages you can use accumulator operators with
+from the Server manual section on
+:manual:`Accumulators </reference/operator/aggregation/#std-label-agg-operators-group-accumulators>`.
+
+.. _java_aggregates_min_n:
+
+MinN
+~~~~
+
+The ``minN()`` builder creates the :manual:`$minN </reference/operator/aggregation/minN/>`
+accumulator which returns data from documents that contain the ``n`` lowest
+values of a grouping.
+
+.. tip::
+
+   The ``$minN`` and ``$bottomN`` accumulators can perform similar tasks.
+   See
+   :manual:`Comparison of $minN and $bottomN Accumulators </reference/operator/aggregation/minN/#comparison-of--minn-and--bottomn-accumulators>`
+   for recommended usage of each.
+
+The following example demonstrates how to use the ``minN()`` method to return
+the lowest three ``imdb.rating`` values for movies, grouped by ``year``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AccumulatorsPickN.java
+   :start-after: // begin minN accumulator
+   :end-before: // end minN accumulator
+   :language: java
+   :dedent:
+
+See the `minN API documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html#minN(java.lang.String,InExpression,NExpression)>`__
+for more information.
+
+.. _java_aggregates_max_n:
+
+MaxN
+~~~~
+
+The ``maxN()`` accumulator returns data from documents that contain the ``n``
+highest values of a grouping.
+
+The following example demonstrates how to use the ``maxN()`` method to
+return the highest two ``imdb.rating`` values for movies, grouped by ``year``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AccumulatorsPickN.java
+   :start-after: // begin maxN accumulator
+   :end-before: // end maxN accumulator
+   :language: java
+   :dedent:
+
+See the `maxN API documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/Accumulators.html#maxN(java.lang.String,InExpression,NExpression)>`__
+for more information.
+
+.. _java_aggregates_first_n:
+
+FirstN
+~~~~~~
+
+The ``firstN()`` accumulator returns data from the first ``n`` documents in
+each grouping for the specified sort order.
+
+.. tip::
+
+   The ``$firstN`` and ``$topN`` accumulators can perform similar tasks.
+   See
+   :manual:`Comparison of $firstN and $topN Accumulators </reference/operator/aggregation/firstN/#comparison-of--firstn-and--topn-accumulators>`
+   for recommended usage of each.
+
+The following example demonstrates how to use the ``firstN()`` method to
+return the first four ``title`` values, based on the order they came into the
+stage, for movies, grouped by ``year``:
+
```

Comment:
issue (non-blocking): This is slightly out of alignment with the way you explain the other examples. I like the way you explain it in others as it comes across more clearly.

```suggestion
The following example demonstrates how to use the ``firstN()`` method to
return the first four movie ``title`` values, based on the order they came into the
stage, grouped by ``year``:

```